app/models/snp_reference.rb

```diff
@@ -3,4 +3,5 @@ class SnpReference < ActiveRecord::Base
   belongs_to :snp
   belongs_to :paper, polymorphic: true
   #validates_presence_of :snp, :paper
+  validates :snp_id, :uniqueness => { :scope => [:paper_id, :paper_type]}
```

Comment:
The hound is supposed to prefer the colon-syntax over the hash-rocket-syntax... Stupid dog.